### ISSUE TITLE
Writing to XLSX only outputs String fields but XLS works fine

### ISSUE THREAD

**Issue by juarezr** [ZuInnoTe/spark-hadoopoffice-ds#27]
When writing to XLSX all int fields are output as 0.

This code works:

```
myDF.select("value1", "value2", "year","month","day")
	.write().format("org.zuinnote.spark.office.excel")
	.option("hadoopoffice.write.mimeType", "application/vnd.ms-excel")
	.option("hadoopoffice.write.header.write", "true")
	.save("hdfs://user/spark/warehouse/output");
```
And outputs:
```
+-----+---+--------+------+
| Name|Age|Children|Salary|
+-----+---+--------+------+
| John| 33|       1| 10000|
| Mary| 43|       9|  1000|
|Peter| 71|       3|  5000|
|James| 22|       1|  2000|
|Homer| 55|       3|  2222|
+-----+---+--------+------+

```
This doesn't works:

```
myDF.select("value1", "value2", "year","month","day")
	.write().format("org.zuinnote.spark.office.excel")
	.option("hadoopoffice.write.mimeType", "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet")
	.option("hadoopoffice.write.header.write", "true")
	.save("hdfs://user/spark/warehouse/output");
```
Neither this works:

```
myDF.select("value1", "value2", "year","month","day")
	.write().format("org.zuinnote.spark.office.excel")
	.save("hdfs://user/spark/warehouse/output");
```
And outputs as:
```
+-----+---+--------+------+
| Name|Age|Children|Salary|
+-----+---+--------+------+
| John| 0 |       0|     0|
| Mary| 0 |       0|     0|
|Peter| 0 |       0|     0|
|James| 0 |       0|     0|
|Homer| 0 |       0|     0|
+-----+---+--------+------+

```
Attached a test case in Java.

**Comment by juarezr** [comment]
Java program for reproducing the issue:
[xlsx-java-test-case.zip](https://github.com/ZuInnoTe/spark-hadoopoffice-ds/files/3453963/xlsx-java-test-case.zip)

**Comment by jornfranke** [comment]
Thx a lot for the detailed description. I will look into it.

**Comment by juarezr** [comment]
Spreadsheet showing the error:
[part-00000-9e7554b4-e704-4592-91d0-ab6b3d628fd1-c000.xlsx](https://github.com/ZuInnoTe/spark-hadoopoffice-ds/files/3453979/part-00000-9e7554b4-e704-4592-91d0-ab6b3d628fd1-c000.xlsx)

**Comment by jornfranke** [comment]
Hmm I get now an idea looking at the .xml of the Excel file. It seems POI is able to write it and read it back correctly, but Excel not. Hence, it was not discovered in the unit tests and I believe this is also a regression that was introduced with 1.3.0 and maybe a different behavior in POI 4.1 (not sure though). I will see for a fix towards the weekend

**Comment by jornfranke** [comment]
another possible workaround for now, can you please set "hadoopoffice.write.lowFootprint" to "true"? I have not try this, but this uses a different writer that may not show this behavior.

**Comment by jornfranke** [comment]
it seems also that it does not happen with all .xlsx files because I can see .xlsx files that work with numbers

**Comment by juarezr** [comment]
Setting `hadoopoffice.write.lowFootprint` didn't change the results.
Also tried to change from native types to object types (int -> Integer) without success:
```
root
 |-- Name: string (nullable = true)
 |-- Age: integer (nullable = true)
 |-- Children: short (nullable = true)
 |-- Salary: long (nullable = true)
```

**Comment by jornfranke** [comment]
Ok i have probably fixed this already and will push out a new version today / tomorrow
It seems only affect Excel itself, because POI reads the value correctly back.

**Comment by jornfranke** [comment]
ok, i have fixed it. I used your program with the patch and the problem disappears and it works as expected. Unfortunately, this had to be tested by looking at it with Excel, because when the processing continues within HadoopOffice it reads/write/reads everything correctly. Now also Excel displays it correctly (the data was btw. contained in the Excel correctly).
I will push 1.3.3 and then you can test. Sorry for the inconvenience

**Comment by jornfranke** [comment]
I published 1.3.3 on Maven

**Comment by juarezr** [comment]
I tested and noticed that the bug is gone.
